packages/edit-post/src/plugins/navigation-edit-menu-item.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	BlockEditorProvider,
+	BlockTools,
+	__unstableBlockToolbarLastItem,
+	__unstableBlockNameContext,
+} from '@wordpress/block-editor';
+import { ToolbarButton, ToolbarGroup } from '@wordpress/components';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { Fragment } from '@wordpress/element';
+import { ReusableBlocksMenuItems } from '@wordpress/reusable-blocks';
+import { __ } from '@wordpress/i18n';
+
+const NavMenuSidebarToggle = () => {
+ 	// Blocks can be loaded into both post and site editors.
+	// We use this to determine which editor we are in so that
+	// we can determine which inspector controls to open.
+	const {
+		isPostEditor,
+	} = useSelect(
+		( select ) => {
+			// eslint-disable-next-line @wordpress/data-no-store-string-literals
+			const editorSelectors = select( 'core/editor' );
+			return {
+				isPostEditor: !! editorSelectors.getEditedPostAttribute( 'type' )
+			};
+		}
+	);
+
+	// eslint-disable-next-line @wordpress/data-no-store-string-literals
+	const { openGeneralSidebar } = useDispatch( 'core/edit-post' );
+	// eslint-disable-next-line @wordpress/data-no-store-string-literals
+	const { enableComplementaryArea } = useDispatch( 'core/interface' );
+	const openBlockInspector = () => {
+		if ( isPostEditor ) {
+			openGeneralSidebar( 'edit-post/block' );
+		} else {
+			enableComplementaryArea( 'core/edit-site', 'edit-site/block-inspector' );
+		}
+	};
```

Comment:
It feels like we should either have:

- normalized the structure of the editors
- create an abstraction to handle this kind of stuff

Not for this PR though...